File: app/src/main/java/peter/util/searcher/view/WebViewContainer.java
package peter.util.searcher.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.ViewParent;
import android.widget.FrameLayout;

import peter.util.searcher.utils.Constants;

/**
 * webView 容器类
 * Created by peter on 2017/1/6.
 */

public class WebViewContainer extends FrameLayout {

    public WebViewContainer(Context context) {
        super(context);
    }

    public WebViewContainer(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public WebViewContainer(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setCurrentView(View view) {
        if (getChildAt(0) != view) {
            removeViewAt(0);
            ViewParent parent = view.getParent();
            if (parent != null) {
                ((ViewGroup) parent).removeAllViews();
            }
            addView(view, 0);
        }
    }

    @Override
    public void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        setMeasuredDimension(getMeasuredWidth(), getMeasuredHeight() + Constants.getActionBarH(getContext()));
    }

}

File: app/src/main/java/peter/util/searcher/view/WebViewContainer.kt
package peter.util.searcher.view

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout

import peter.util.searcher.utils.Constants

/**
 * webView 容器类
 * Created by peter on 2017/1/6.
 */

class WebViewContainer : FrameLayout {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    fun setCurrentView(view: View) {
        removeViewAt(0)
        addView(view, 0)
    }

    fun setCurrentView(viewId: Int): View {
        removeViewAt(0)
        val factory = LayoutInflater.from(context)
        val mView = factory.inflate(viewId, this, false)
        addView(mView, 0)
        return mView
    }

    public override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        setMeasuredDimension(measuredWidth, measuredHeight + Constants.getActionBarH(context))
    }

}
